File: 3. Longest Substring Without Repeating Characters.java
public class Solution {
    
    int[] index = new int[256];
    public int lengthOfLongestSubstring(String s) {
        int maxLength = 0;
        int len = 0;
        for (int i = 0; i < 256 ; i++) {
            index[i] = -1;
        }
        for (int i = 0; i < s.length() ; i++) {
            char c = s.charAt(i);
            if(index[c] >= 0) {
                if(i - index[c] > len)
                    len++;
                else
                    len = i - index[c];
            } else {
                len++;
            }
            index[c] = i;
            if(len > maxLength)
                maxLength = len;
        }
        return maxLength;
    }
}

File: 3. Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int, int> map;
        int len = 0, max = 0;
        string::iterator it = s.begin();
        
        for(int i = 0; it < s.end(); it++, i++) {
            if(map.find(*it - 'a') != map.end()) {
                len = i - map[*it - 'a'] > len ? len+1 : i - map[*it - 'a'];
            } else {
                len++;
            }
            map[*it - 'a'] = i;
            max = max > len ? max : len;
        }
        
        return max;
    }
};